students_db/src/main/java/application/dao_impls/DisciplineDaoImpl.java

```diff
@@ -1,84 +1,69 @@
 package application.dao_impls;
 
 import application.dao_interfaces.DisciplinesDao;
-import application.models.Disciplines;
+import application.models.Discipline;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
 import org.springframework.jdbc.core.JdbcTemplate;
-import org.springframework.jdbc.core.RowMapper;
 import org.springframework.stereotype.Repository;
 
-import java.sql.ResultSet;
-import java.sql.SQLException;
 import java.util.List;
 
 @Repository
 @Qualifier("disciplineDao")
 public class DisciplineDaoImpl implements DisciplinesDao {
-    private String sql;
 
     @Autowired
     private JdbcTemplate jdbcTemplate;
 
     @Override
-    public void insertDiscipline(Disciplines disciplines) {
-        sql = "insert into discipline (discipline_id, discipline_name) values (?, ?);";
-        jdbcTemplate.update(sql, disciplines.getId(), disciplines.getName());
+    public int insertDiscipline(Discipline discipline) {
+        String sql = "insert into disciplines (discipline_id, discipline_name) values (?, ?);";
```

Comment:
you should return id which generate database(or return discipline with id from database)